can/interfaces/socketcan/socketcan.py

```diff
@@ -341,22 +355,38 @@ def stop(self):
         stopframe = build_bcm_tx_delete_header(self.can_id_with_flags, self.flags)
         send_bcm(self.bcm_socket, stopframe)
 
-    def modify_data(self, message):
-        """Update the contents of this periodically sent message.
+    def modify_data(self, messages):
+        """Update the contents of the periodically sent messages.
+
+        Note: The messages must all have the same
+        :attr:`~can.Message.arbitration_id` like the first message.
 
-        Note the Message must have the same :attr:`~can.Message.arbitration_id`
-        like the first message.
+        Note: The number of new cyclic messages to be sent must be equal to the
+        original number of messages originally specified for this task.
+
+        :param Union[List[can.Message], tuple(can.Message), can.Message] messages:
```

Comment:
nit: I'd do this with typing.Sequence: `Union[Sequence[can.Message], can.Message]`

Reply:
Or [typing.Iterable](https://docs.python.org/3/library/typing.html#typing.Iterable)?

Reply:
@felixdivo correct me if I'm wrong, but I don't think you can get the length of an `Iterable` without actually performing the iteration and storing the result.

If want to support an `Iterable`, do we want to deal with buffering potentially unbounded input? Or implement specific bounded iterators for each interface that only iterates over the maximum number of elements the interface task can support, and errors otherwise?

Reply:
Ah I see. We could wrap it in a list if it's not already a sequence. Implementing something like that ourselves seems overkill. If you want, stay with what you have or a sequence.